garak/probes/base.py

```diff
@@ -183,6 +183,42 @@ def _mint_attempt(
         self, prompt=None, seq=None, notes=None, lang="*"
     ) -> garak.attempt.Attempt:
         """function for creating a new attempt given a prompt"""
+        turns = []
+        if isinstance(prompt, garak.attempt.Conversation):
+            try:
+                # only add system prompt if the prompt does not contain one
+                prompt.last_message("system")
+                turns = prompt.turns
+            except ValueError as e:
+                turns.append(prompt.turns)
+        elif hasattr(self, "system_prompt") and self.system_prompt:
+            turns.append(
+                garak.attempt.Turn(
+                    role="system",
+                    content=garak.attempt.Message(
+                        text=_config.run.system_prompt, lang=lang
+                    ),
+                )
+            )
+        if isinstance(prompt, str):
+            turns.append(
+                garak.attempt.Turn(
+                    role="user", content=garak.attempt.Message(text=prompt, lang=lang)
+                )
+            )
+        elif isinstance(prompt, garak.attempt.Message):
+            turns.append(garak.attempt.Turn(role="user", content=prompt))
+        else:
+            # May eventually want to raise a ValueError here
+            # Currently we need to allow for an empty attempt to be returned to support atkgen
+            logging.warning("No prompt set for attempt in %s" % self.__class__.__name__)
+
+        if len(turns) > 0:
+            prompt = garak.attempt.Conversation(
+                turns=turns,
+                notes=notes,
+            )
+
```

Comment:
when would we want to permit a prompt with empty `Conversation`? (distinct from prompt of `Conversation` with `Turn` of empty string)

Reply:
I don't thing empty turns is a thing, this guard is simply to avoid changing the input param `prompt` if the local helper variable `turns` was never populated.  Again this may be possible to removed if/when we validate that all `prompt` values are `Conversation` objects.